### PR TITLE
[Poetry] Use react-select for poem dropdown

### PR DESCRIPTION
**Pull request by ajpal**
Use `react-select` instead of HTML `<select>` for the poem dropdown. Driving reason for this is that we want the change handler to fire even if the same option is selected again.

Before:
image

Oct-14-2021 14-55-09

After:
image

Oct-14-2021 14-56-03

## Links

[slack thread](https://codedotorg.slack.com/archives/C0T0PNR0D/p1633564072024700)

## PR Checklist:

- [ ] Tests provide adequate coverage
- [ ] Privacy and Security impacts have been assessed
- [ ] Code is well-commented
- [ ] New features are translatable or updates will not break translations
- [ ] Relevant documentation has been added or updated
- [ ] User impact is well-understood and desirable
- [ ] Pull Request is labeled appropriately
- [ ] Follow-up work items (including potential tech debt) are tracked and linked

### CODE REVIEW

File: apps/style/JsDebuggerUi.scss [hunk 1]
```diff
@@ -199,11 +200,3 @@ $debug-header-border-width: 1px;
   background-position: 0px 21px;
   margin-right: 3px;
 }
-
-.Select-input {
-  height: 28px !important;
-}
-
-.Select-arrow-zone {
```
Comment (ajpal) [inline apps/style/JsDebuggerUi.scss:207]: This comes from this very old commit: https://github.com/code-dot-org/code-dot-org/commit/1f5f0045cd7c9752e3362277ebadda4b8453c14a

And from searching the codebase + searching the elements on the page in gamelab, I don't think it's in use anymore, so I think it's safe to remove
image

Comment (maddiedierker) [reply apps/style/JsDebuggerUi.scss:207]: 🎉 🎉 🎉  yay for being able to delete `!important`!

**Review by maddiedierker — APPROVED** [review]

**Comment by breville** [comment]
Just curious: what is this like in iOS Safari?  I'm curious whether it maybe bypasses the use of the browser's native select?  Tangentially curious whether that affects accessibility...

**Comment by ajpal** [comment]
Looks good in Safari:
image

**Comment by breville** [comment]
Ideally, when we offer dropdowns, they'd always be native to the browser, like our language one seems to be, seen here on an iPhone:

Image from iOS

afaict, using react-select gets us a custom implementation of a dropdown:

Image from iOS (1)

that said, it seems to work alright, though I'm curious about screenreaders etc. still.  

And, at any rate, we already use non-native dropdowns elsewhere, so there is precedent for a non-native approach, even if it's not ideal:

Image from iOS (2)

**Pull request merged**